napari/_qt/widgets/magic_gui_dask_settings.py

```diff
@@ -0,0 +1,18 @@
+from magicgui import magic_factory
+
+from napari._qt.widgets.magic_gui_base_widget import BaseMagicSetting
+
+
+@magic_factory(
+    auto_call=True, layout='horizontal', cache={'min': 0, 'max': 20}
+)
+def dask_settings(dask_enabled=True, cache=15.0) -> dict:
+    """Create magic gui function GUI with checkbox and spinbox for dask settings."""
+    return {'enabled': dask_enabled, 'cache': cache}
+
+
+class MagicDaskSettingsWidget(BaseMagicSetting):
+    """Class for use in json schema widget builder for dask settings."""
+
+    def get_mgui(self):
+        return dask_settings()
```

Comment:
@tlambert03 The only implementation of get_mgui is here. MAy you suggest what type of annotation should be used?

Reply:
Dask_settings appears to be a FunctionGui (which is a Container, which is why this works), but you'd have to ask @ppwadhwa whether the design of this abstract class demands that or not

Reply:
@tlambert03 @Czaki I am not sure.  In the end, I just want a widget with a checkbox and spinbox in a horizontal layout to be added to the preferences dialog.  I am trying to use MagicGui to do this, and at the same time trying to fit it all into the json schema builder so it can work the same as every other widget in the preferences dialog.  Should I be doing things differently?  I would appreciate any suggestions on how to move forward.

Reply:
This widget is fine.  The issue is over in the base widget, where you've created what appears to be an abstract class for "any generic widget that magicgui might provide"... but then you've used a private attribute specifically from magicgui Containers. So, while this works fine for your one dask widget, someone might reasonably assume that they can use your base class for _any_ magicgui widget (even those which are not containers) and then your setTooltip method will raise an exception

Reply:
@tlambert03 thank you for the explanation.  I see the problem now.